### PR TITLE
Add attributes section to docc index

### LINKED ISSUE #1973: Create new section for attribute nodes in docc index
[SwiftSyntax.md](https://github.com/apple/swift-syntax/blob/main/Sources/SwiftSyntax/Documentation.docc/generated/SwiftSyntax.md) has sections for expression, declaration etc. nodes. A large portion of the nodes in the Miscellaneous section concern attributes. We should have a section for all the attributes nodes. All nodes from the [AttributeNodes.swift](https://github.com/apple/swift-syntax/blob/main/CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift) file should end up in that section.

### PR DESCRIPTION
**Pull request by Matejkob**
Resolves #1973

### CODE REVIEW

**Review by ahoppen — APPROVED** [review]
Oh, that was easy 👍🏽

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SwiftSyntaxDoccIndex.swift [hunk 1]
```diff
@@ -59,6 +59,8 @@ let nodesSections: String = {
       })
   )
 
+  addSection(heading: "Attributes", types: ATTRIBUTE_NODES.map(\.kind.syntaxType.description))
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SwiftSyntaxDoccIndex.swift:62]: Ah, I do have one comment after all. Could we sort the types alphabetically?

Comment (Matejkob) [reply CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SwiftSyntaxDoccIndex.swift:62]: Great idea! I've been also wondering about sorting the sections in alphabetical order. What do you think about that?

Comment (ahoppen) [reply CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SwiftSyntaxDoccIndex.swift:62]: I thought the other sections are alphabetically sorted, with the exception that the base node (e.g. `ExprSyntax`), its protocol type and the corresponding missing node are first, which makes sense to me.

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Pull request merged**